plugins/auth-backend/src/lib/passport/PassportStrategyHelper.ts

```diff
@@ -57,6 +58,26 @@ export const makeProfileInfo = (
     }
   }
 
+  const customObjects: any[] = [];
+  if (customAttributes && idToken) {
+    try {
+      const decoded: Record<string, string> = jwtDecoder(idToken);
+      customAttributes.forEach(att => {
```

Comment:
What are you using this for, and how do you consume it in the frontend?

Do you think adding a method for passing in your own profile transform from the backend package might be even better? There's some very loose coupling between the frontend and backend going on here.

If this is for profile-like information we do want to be able to provide that through identities in the catalog instead, although that's further down the line.

Reply:
We were consuming this via the user profile.

Passing your own profile transform might be better. If you want to consume the backend or auth-backend packages as node_modules how would you pass in your own function? I went with this solution because it would allow you to specify what you want from the token without modifying and plugins/packages.

This would be for profile-like information.

Reply:
It would require this to be completed: https://github.com/spotify/backstage/issues/1984

With that you'd essentially be able to pass in your own OAuth provider, with its own profile transform.

And alright, so we're really looking to make profile information available through user entities, but I don't think we can have that streamlined enough soon enough for it to be the only way to access more profile information.

Reply:
It's looks like that issue is similar to mine, but requires more custom setup. I was trying to go with the simplest option and minimal number of changes that would solve the issue of needing more info that is contained in id token then just email and pic.  It would definitely be beneficial to make that information available through user entities, I just didn't think that was ready yet.